contracts/mainnet/connectors/permit_erc20/main.sol

```diff
@@ -0,0 +1,73 @@
+pragma solidity ^0.7.6;
+pragma experimental ABIEncoderV2;
+
+import {TokenInterface, MemoryInterface} from "../../common/interfaces.sol";
+import {Stores} from "../../common/stores.sol";
+import {ERC20_functions, ERC20_dai_functions} from "./interface.sol";
+//import {Helpers} from "./helpers.sol";
+import {Events} from "./events.sol";
+
+/**
+ * @title permit_erc20.
+ * @dev Adding permit functionality to ERC_20.
+ */
+
+
+contract permit_erc20 {
+    address private immutable daiAddress = 0x6B175474E89094C44Da98b954EedeAC495271d0F; // dai has a different implementation for permit
+
+    /**
+     * @notice ERC20_Permit functionality
+     * @dev Adding permit functionality to ERC_20.
+     * @param _asset The address of the token to call.(For AAVE Token : 0x7Fc66500c84A76Ad7e9c93437bFc5Ac33E2DDaE9)
+     * @param _owner The public of the user which wants to permit the user to take funds.(Ex: - 0x3Fc046bdE274Fe8Ed2a7Fd008cD9DEB2540dfE36 )
+     * @param nonce The nonce of the user(Neede only if asset is DAI)  //can add helper here
```

Comment:
@harshit-wq what do you mean by `helper`?

Reply:
We get nonce by interacting with the token deployed contract, so i was thinking if we could provide user with the facility to not worry about getting it himself and we get the nonce for the user ourselves by using some kind of helper function. Like we could just remove this parameter.

Reply:
@harshit-wq This nonce is specific to which permit contract you are interacting with, am I right?

Reply:
Yes @pradyuman-verma For example, we need nonce for dai. So we fetch this nonce from dai contract.

Reply:
Okay, then let's add a helper function that returns the NONCE of the user.

Reply:
Also, can you resolve conflicts?

Reply:
Done